### PR TITLE
Search component - event/search

### PR DESCRIPTION
**Pull request by cjkeilig**
This PR should get a good start on #101 .

- Restyles the right header buttons when logged in
- Adds a search button that changes the header to a search bar when clicked
- Adds a search component (route event/search) that display a grid of events filtered by the text in the search bar or in the url parameter.
- The filter applies to text in the event name, date, and type fields

### CODE REVIEW

File: src/frontend/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -13,16 +12,12 @@ import { UsersComponent } from './components/users/users.component';
...
-      { path: 'notauthorized', component: NotauthorizedComponent},
-      { path: 'users', component: UsersComponent, canActivate: [AuthGuard]},
-    ]},
-{ path: '**', component: NotfoundComponent },
+  { path: 'login', component: LoginComponent},
+  { path: 'register', component: RegisterComponent},
+  { path: 'home', component: HomeComponent },
+  { path: 'notauthorized', component: NotauthorizedComponent},
+  { path: 'users', component: UsersComponent, canActivate: [AuthGuard]},
+  { path: '**', component: NotfoundComponent },
```
Comment (codykaup) [inline src/frontend/src/app/app-routing.module.ts]: I'm curious why routing is changing this drastically for adding a search bar. 🤔

File: src/frontend/src/app/app.component.html [hunk 2]
```diff
@@ -1 +1,2 @@
-<router-outlet></router-outlet>
+<app-header></app-header>
```
Comment (codykaup) [inline src/frontend/src/app/app.component.html]: This piece was handled by the `LayoutComponent` so I'm wondering what we're gaining by removing it.

**Comment by codykaup** [comment]
A comment for anybody else that reviews this PR: I had some caching fun when adding a new event. It added properly but the home screen didn't show until I did a hard refresh. We may want to add a story to adjust this cache setting in the future.

File: src/frontend/src/app/app-routing.module.ts (hunk 1 above)
Comment (cjkeilig) [reply src/frontend/src/app/app-routing.module.ts]: I was having an issue at one point, got confused, and changed this. 
I don't know why we have a 'LayoutComponent' and only a `router-outlet` in the app.component.html. 

If there is a good reason, then by all means let's keep it, but it is confusing to me.

**Comment by cjkeilig** [comment]
Thanks for the review on this Cody, we can comment out the caching so it can be turned on in the future if we want.
I think you are right that we need a general purpose auto-complete control that can be passed an array of strings and has an animation and drop down. 
 
Again, I don't know why we _need_ a `LayoutComponent` as it is implicit from the main app.component.html.

**Pull request merged**